### PR TITLE
Patch icu external project

### PR DESCRIPTION
**Pull request by mryzhov**
Closes issue https://github.com/openvinotoolkit/openvino_tokenizers/issues/171

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -142,11 +142,18 @@ function(ov_tokenizers_link_sentencepiece TARGET_NAME)
 endfunction()
 
 if(BUILD_FAST_TOKENIZERS)
+  set(EXTERNAL_PROJECT_SOURCE_DIR ${CMAKE_BINARY_DIR}/_deps/fast_tokenizer/src)
+  set(EXTERNAL_PROJECT_BINARY_DIR ${CMAKE_BINARY_DIR}/_deps/fast_tokenizer/build)
+  set(EXTERNAL_PROJECT_SUBBUILD_DIR ${CMAKE_BINARY_DIR}/_deps/fast_tokenizer/sub-build)
```
Comment (ilya-lavrenov) [inline src/CMakeLists.txt:147]: why do we need non-default directories?

Comment (mryzhov) [reply src/CMakeLists.txt:151]: We need to know the root source path of the external project to apply properly the patch (using git apply)

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -41,12 +41,16 @@ set(CMAKE_POSITION_INDEPENDENT_CODE ON)
...
-cmake_dependent_option(BUILD_FAST_TOKENIZERS "Compile core_tokenizers instead of downloading prebuilt library. The option is hidden and forced to ON if _GLIBCXX_USE_CXX11_ABI=0"
-  OFF "NOT _GLIBCXX_USE_CXX11_ABI=0 IN_LIST OPENVINO_RUNTIME_COMPILE_DEFINITIONS" ON)
 if("_GLIBCXX_USE_CXX11_ABI=0" IN_LIST OPENVINO_RUNTIME_COMPILE_DEFINITIONS)
   set(USE_ABI0 ON CACHE BOOL "Set -D_GLIBCXX_USE_CXX11_ABI to 0 for fast_tokenizers")
 endif()
 
+option(BUILD_FAST_TOKENIZERS "Compile core_tokenizers instead of downloading prebuilt library. The option is forced to ON if _GLIBCXX_USE_CXX11_ABI=0" ${USE_ABI0})
+# py_build_cmake doesn't support environment cmake options
+if(DEFINED ENV{BUILD_FAST_TOKENIZERS})
+  set(BUILD_FAST_TOKENIZERS $ENV{BUILD_FAST_TOKENIZERS})
```
Comment (ilya-lavrenov) [inline src/CMakeLists.txt:51]: I suppose it should be cmake option or least let's set value to cache

Comment (mryzhov) [reply src/CMakeLists.txt]: removed

**Review by apaniukov — APPROVED** [review]

**Review by ilya-lavrenov — APPROVED** [review]

**Pull request merged**